storage/storage_tests/storage_tests.cpp

```diff
@@ -1227,4 +1221,39 @@ UNIT_TEST(StorageTest_ParseStatus)
   TEST_EQUAL(StatusAndError(NodeStatus::Downloading, NodeErrorCode::NoError),
              ParseStatus(Status::EDownloading), ());
 }
+
+UNIT_TEST(StorageTest_ForEachInSubtree)
+{
+  Storage storage(kSingleMwmCountriesTxt, make_unique<TestMapFilesDownloader>());
+
+  TCountriesVec leafVec;
+  auto const forEach = [&leafVec](TCountryId const & descendantCountryId, bool expandableNode)
+  {
+    if (!expandableNode)
+      leafVec.push_back(descendantCountryId);
+  };
+  storage.ForEachInSubtree(storage.GetRootId(), forEach);
+
+  TCountriesVec const expectedLeafVec = {"Abkhazia", "Algeria_Central", "Algeria_Coast", "South Korea_South"};
+  TEST_EQUAL(leafVec, expectedLeafVec, ());
+}
+
+UNIT_TEST(StorageTest_CalcLimitRect)
+{
+  Storage storage(COUNTRIES_FILE);
```

Comment:
Looks like you misunderstood my question.
Storage now have two constructors, the first one is used in prod, the second one is for testing purposes only (it's claimed in a comment). Both constructors take std::string as a first argument and have default values for the second argument. When I'm looking at your code, I do not understand - first or second constructor is called here. It doesn't matter, COUNTRIES_FILE or COUNTRIES_MIGRATE_FILE should be used here, because in any case they are strings. That's why I'm asking you to remove testing constructor and create a factory method instead, like

```c++
static Storage Storage::MakeForTesting(
    string const & referenceCountriesTxtJsonForTesting,
    unique_ptr<MapFilesDownloader> mapDownloaderForTesting);
```
It should not be too hard.